spec/src/main/asciidoc/chapters/language/expressions.adoc

```diff
@@ -423,13 +423,32 @@ Within the pattern, `_` matches any single character, and `%` matches any sequen
 
 The equality and inequality operators are `=`, `&lt;&gt;`, `&lt;`, `&gt;`, `&lt;=`, `&gt;=`.
 
+The operands of an equality or inequality operator must have the same type,
+ and it must be an atomic type considered _comparable_ by the implementation of Jakarta Query.
+Any type with a <<natural-order,natural order>> must be treated as a comparable type, and the equality and inequality operators must be interpreted in a way which is consistent with the natural order.
+
+[cols="15,~"]
+|===
+| Operator | Interpretation in terms of natural order
+
+| `x < y` | Satisfied if `x` occurs before `y` in the natural order
+| `x > y` | Satisfied if `x` occurs after `y` in the natural order
+| `x <= y` | Satisfied if `x>y` is not satisfied
+| `x >= y` | Satisfied if `x<y` is not satisfied
+| `x = y` | Satisfied if neither `x>y` nor `x<y` is satisfied
```

Comment:
I saw your later TODO about nulls, where we will need to avoid the possibility that neither `null < y` nor `null > y` are satisfied implies that `null = y` is satisfied.  Would the following work?
```suggestion
| `x = y` | Satisfied if `x` and `y` occur at identical position in the natural order
```

Reply:
@njr-11 I've added language clarifying what's going on with `null`.

Actually I had not forgotten about it, I had previously left it as a TODO:

> **TODO** what can we say in general about null values?

This is all a bit complicated to deal with right now because of the multiple open pull requests dealing with interlocking concerns. So I would like to just get all this shit integrated and then revisit the everything later when we can see it all in front of us at once.